src/core/control/tools/StrokeHandler.cpp

```diff
@@ -89,8 +87,12 @@ void StrokeHandler::paintTo(const Point& point) {
                 this->stroke->setLastPressure(point.z);
                 this->firstPointPressureChange = true;
 
-                // Paint the dot for the user to see
-                this->paintDot(endPoint.x, endPoint.y, this->stroke->getWidth() * point.z);
+                double width = this->stroke->getWidth() * point.z;
+                if (!this->fullRedraw) {
+                    // Paint the dot for the user to see
+                    this->paintDot(crMask, endPoint.x, endPoint.y, width);
+                }
+                this->redrawable->repaintRect(endPoint.x - 0.5 * width, endPoint.y - 0.5 * width, width, width);
```

Comment:
Do we need to repaint around the end point if `fullRedraw` is true? The endpoint is only drawn if false.

Reply:
`repaintRect` triggers the call to `draw`, where the `paintDot` will be called in the case we don't use the mask.
I'll add a comment.